In [1]:
#import libraries
import numpy as np
import pandas as pd

In [9]:
arr = np.array([10, 20, 30, 40, 50, 60, 70])
print(arr)
print("\nAverage number", arr.mean())

[10 20 30 40 50 60 70]

Average number 40.0


In [11]:
#indexing
print(arr[3])
print(arr[-2])
print(arr[6])
print(arr[-4])

40
60
70
40


In [13]:
#slicing
print(arr[1:6])
print(arr[6:])
print(arr[:7])
print(arr[-6:-1])
print(arr[7:])
print(arr[-1:-6])

[20 30 40 50 60]
[70]
[10 20 30 40 50 60 70]
[20 30 40 50 60]
[]
[]


In [14]:
#2D indexing and slicing
matrix = np.array([
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
])

print(matrix[0, 1])
print(matrix[1, :])
print(matrix[1:2, 2])
print(matrix[2, :2])
print(matrix[1:2, 0:2])

2
[4 5 6]
[6]
[7 8]
[[4 5]]


In [18]:
#change 1D array to 2D array
arr2 = np.arange(1, 13)
reshape = arr2.reshape(3, 4)

print("Original array:", arr2)
print("\nReshape array:", reshape)

#change 2D to 1D array
flatten = reshape.flatten()
print("Flatten:", flatten)

Original array: [ 1  2  3  4  5  6  7  8  9 10 11 12]

Reshape array: [[ 1  2  3  4]
 [ 5  6  7  8]
 [ 9 10 11 12]]
Flatten: [ 1  2  3  4  5  6  7  8  9 10 11 12]


In [20]:
#working with basic mathematics operators
arr3 = np.array([1, 2, 3, 4, 5])
arr4 = np.array([10, 20, 30, 40, 50])

#basic operators
print(arr3 + arr4)
print(arr3 - arr4)
print(arr3 * arr4)
print(arr3 / arr4)

#scaler operator
print(arr4 ** 2)
print(arr4 // 2)

[11 22 33 44 55]
[ -9 -18 -27 -36 -45]
[ 10  40  90 160 250]
[0.1 0.1 0.1 0.1 0.1]
[ 100  400  900 1600 2500]
[ 5 10 15 20 25]


In [22]:
#working with dot product
matrix1 = np.array([
    [1, 2],
    [3, 4]
])

matrix2 = np.array([
    [5, 6],
    [7, 8]
])

dot = np.dot(matrix1, matrix2)
print("Dot Product\n", dot)

multiply = matrix1 * matrix2
print("Multiply\n", multiply)

Dot Product
 [[19 22]
 [43 50]]
Multiply
 [[ 5 12]
 [21 32]]


In [3]:
#30 DAYS DAILY STEPS TRACKER
np.random.seed(42) #reproducibility
days = np.arange(1, 31)
steps = np.random.randint(3456, 12034, size=30)

print("Days:", days)
print("Steps:", steps)

Days: [ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24
 25 26 27 28 29 30]
Steps: [10726  4316  8846  8647  9190  9721  3922  7882  9034 11778  5141  4225
 10405  5889  8767  8507  9876  4640  8011  6841  9852  6014 11305  5503
  6203  3645  6190  6461  8114  5355]


In [4]:
data = pd.DataFrame({
    "Days": days,
    "Steps": steps
})

data

,Days,Steps
0,1,10726
1,2,4316
2,3,8846
3,4,8647
4,5,9190
5,6,9721
6,7,3922
7,8,7882
8,9,9034
9,10,11778


In [5]:
#basic stats 
mean_step = np.mean(steps)
std_step = np.std(steps)
print("Average step:", mean_step)
print("Median step:", np.median(steps))
print("Minimum step:", np.min(steps))
print("Maximum step:", np.max(steps))
print("Standard deviation step:", std_step)

Average step: 7500.2
Median step: 7946.5
Minimum step: 3645
Maximum step: 11778
Standard deviation step: 2309.770715893679


In [9]:
#goal analysis for days exceeding 10,000 steps
goal = np.where(steps > 10000)[0] + 1
print(goal)

[ 1 10 13 23]


In [11]:
#analysis for weekly steps
weeks = np.array_split(steps, 4)
print("Week total analysis:")
week_total = [print(week.sum()) for week in weeks]
print("\nWeek average analysis:")
week_avg = [print(week.mean()) for week in weeks]

Week total analysis:
63250
63746
56539
41471

Week average analysis:
7906.25
7968.25
8077.0
5924.428571428572


In [15]:
#anomaly detection
lower_bound = mean_step - 2 * std_step
upper_bound = mean_step + 2 * std_step

print(round(lower_bound, 2))
print(upper_bound)

anomaly = np.where((steps < lower_bound) | (steps > upper_bound))[0] + 1
print('Anomaly detection (low/high):', anomaly)

2880.6585682126415
12119.741431787359
Anomaly detection (low/high): []


The average daily steps is 7,500, meaning that on a typical day the person walk around seventy-five hundred steps.

The median is 7,946, a bit higher than the mean, which suggests most days are on the more active side.

The lowest day is 3,645 steps, likely a low-movement day, while the highest day reaches 11,778, marking the most active day of the month. The standard deviation is 2,309, showing that the activity varies a lot, some days are very active, others much lighter.